project_functional_block/models/project_functional_block.py

```diff
@@ -0,0 +1,146 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    OpenERP, Open Source Management Solution
+#    Copyright (C) 2010 Akretion LDTA (<http://www.akretion.com>).
+#
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp import models, fields, api
+
+
+class ProjectFunctionalBlock(models.Model):
+    _name = 'project.functional.block'
+    _description = 'Functional block to organize projects tasks'
+    _order = 'parent_id desc, name'
+    _rec_name = 'complete_name'
+    _parent_store = True
+    _parent_order = 'name'
+
+    # Columns
+    name = fields.Char('Name', size=64, required=True, translate=True)
+    code = fields.Char('Code', size=8)
+    complete_name = fields.Char(
+        string='Complete Name', compute='_compute_complete_name')
+    description = fields.Text('Description', translate=True)
+    parent_id = fields.Many2one(
+        'project.functional.block', 'Parent Block',
+        ondelete='cascade', index=True)
+    parent_left = fields.Integer('Left Parent', index=True)
+    parent_right = fields.Integer('Right Parent', index=True)
+    tasks_count = fields.Integer(string='Tasks Count', compute='_count_tasks')
+    child_fblocks_count = fields.Integer(
+        string='Child Blocks Count', compute='_count_child_fblocks')
+
+    @api.multi
+    def _compute_complete_name(self):
```

Comment:
I think this can be implemented with a single line:

    self.complete_name = '%s / %s' % (self.parent_id.complete_name, self.name)